Program.cs

```diff
@@ -4,33 +4,64 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using CMP1903M_Assessment_1.Debugging;
 
 namespace CMP1903M_Assessment_1
 {
     class Program
     {
         static void Main()
         {
-            //Local list of integers to hold the first five measurements of the text
-            List<int> parameters = new List<int>();
-
-            //Create 'Input' object
-            //Get either manually entered text, or text from a file
-
-
-            //Create an 'Analyse' object
-            //Pass the text input to the 'analyseText' method
-
-
-            //Receive a list of integers back
-
-
-            //Report the results of the analysis
-
-
-            //TO ADD: Get the frequency of individual letters?
-
-           
+            Console.WriteLine("Main Menu");
+            Console.WriteLine("What type of input would you like to use?\n1 Manual Text Entry\n2 Text File");
+            string input = Console.ReadLine();
+            int selection = 0;
+            if (int.TryParse(input, out selection))
+            {
+                Analyse analysis = null;
+                switch (selection)
+                {
+                    case 1:
+                        // manual text input
+                        analysis = new Analyse(Input.ManualTextInput());
+                        break;
+                    case 2:
+                        analysis = new Analyse(Input.FileTextInput());
+                        break;
+                    default:
+                        Debug.LogError($"Unrecognised option {selection}");
+                        Main();
+                        break;
+                }
+                if (analysis != null)
+                {
+                    analysis.AnalyseText();
+                    Console.WriteLine("Report generated. Would you like to: \n1. View in Console (Default)\n2. View in File");
+                    // find out whether the user wants to report to console, file, or both
+                    string response = Console.ReadLine() ?? "1"; // Default to view in console, if null
+                    char[] responseChars = response.Trim().ToCharArray();
+                    bool console = false;
+                    bool file = false;
+                    foreach (char c in responseChars)
+                    {
+                        if (c == '1')
+                            console = true;
+                        else if (c == '2')
+                            file = true;
+                        else
+                            Debug.LogError($"Unrecognised option: {c.ToString()}");
+                    }
+                    if (!console && !file) { console = true; }; // ensures that the report will go to console if selects neither

```

Comment:
This line may be simplified:
```suggestion
                    console = !console && !file // ensures that the report will go to console if selects neither

```